streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -143,22 +154,27 @@ public void register(final StateRestorer restorer) {
             TopicPartition topicPartition = entry.getKey();
             Long offset = entry.getValue();
             final StateRestorer restorer = stateRestorers.get(topicPartition);
-            if (restorer.checkpoint() >= offset) {
-                restorer.setRestoredOffset(restorer.checkpoint());
-            } else if (restorer.offsetLimit() == 0) {
-                restorer.setRestoredOffset(0);
-            } else {
-                needsRestoring.put(topicPartition, restorer);
-                final Long endOffset = endOffsets.get(topicPartition);
-                restorer.restoreStarted(restorer.startingOffset(), endOffset);
+            // might be null as has was initialized in a previous invocation.
+            if (restorer != null) {
```

Comment:
I am not sure why restorer can ever be null here: the `topicPartition` is from the `endOffsets` entry which was computed based on the passed in `stateRestorers`. So how can `stateRestorers.get(topicPartition)` return null?